source/grid/grid_tools.cc

```diff
@@ -1606,6 +1606,45 @@ namespace GridTools
   }
 
 
+  template <class MeshType>
+  std::vector<typename MeshType::cell_iterator>
+  compute_cell_halo_layer_on_level
```

Comment:
i think it would be better if you
1. try to re-use  `compute_active_cell_halo_layer()` (maybe extend it or maybe move the implementation inside internal function and call it from two places). Either way, it look slike there is a fair amount of code duplication here. There only difference is `begin_active()/end()` vs `begin(level) /end(level)`.
2. add a predicate here `const std_cxx11::function<bool (const typename MeshType::active_cell_iterator &)> &predicate` instead of hard-coding comparison of subdomain id.